app-emulation/wa-linux-agent/wa-linux-agent-2.6.0.2-r1.ebuild

```diff
@@ -1,36 +1,40 @@
 # Copyright (c) 2014 CoreOS, Inc.. All rights reserved.
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=7
+EAPI=8
+
+# Don't use DISTUTILS_USE_PEP517=setuptools because this installs
+# everything inside /usr/lib/pythonX_Y/site-packages, even files that
+# ought to be put into /etc or /sbin.
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit distutils-r1
 
 DESCRIPTION="Windows Azure Linux Agent"
 HOMEPAGE="https://github.com/Azure/WALinuxAgent"
-KEYWORDS="amd64 arm64"
 SRC_URI="${HOMEPAGE}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="Apache-2.0"
+KEYWORDS="amd64 arm64"
 SLOT="0"
 IUSE=""
+RESTRICT=""
 
-# Depending on specific version of python-oem allows us to notice when
-# we update the major version of python and then to make sure that we
-# install the package in correctly versioned site-packages directory.
-DEP_PYVER="3.10"
-
-RDEPEND="
-dev-lang/python-oem:${DEP_PYVER}
-dev-python/distro-oem
+BDEPEND="
+	dev-python/distro
+"
+RDEPEND="${DEPEND}
 "
 
 S="${WORKDIR}/WALinuxAgent-${PV}"
 
-src_install() {
-	into "/usr/share/oem"
-	dobin "${S}/bin/waagent"
-
-	insinto "/usr/share/oem/python/$(get_libdir)/python${DEP_PYVER}/site-packages"
-	doins -r "${S}/azurelinuxagent/"
+PATCHES=(
+    "${FILESDIR}/0001-flatcar-changes.patch"
+)
 
-	insinto "/usr/share/oem"
+src_install() {
+	distutils-r1_src_install
+	rm -f "${D}/etc/waagent.conf"
+	insinto /etc
```

Comment:
Where will this file end?

Reply:
That's a good question. This won't be a part of the generic production image and probably also not a part of sysext image (since we only take files from `/usr` into the raw image), so maybe I should actually be putting it in `/usr/share/flatcar/etc`. Or (better IMO to keep packages generic) will need an extra logic for the oem-sysexts, that moves `/etc` to `/usr/share/flatcar/etc` before creating the raw image. I think this could work if `/etc` is set up after sysexts, but not really sure if it works like that now - I have a bit of a chicken-and-egg feeling here…

Reply:
If we know that this won't be written to, I would rather put this file under `/usr/share/waagent/` or one of the paths under `/usr` that this waagent already uses. But if that doesn't work I would we could split this up into two ebuild files by creating a new ebuild file for the generic image that installs only this file under `/etc` and our existing logic will move it to `/usr/share/flatcar/etc`.

Reply:
We can also preserve the current behavior and copy it to `/etc` if it does not exist (`ExecStartPre` because tmpfiles won't pick the rule up)

Reply:
I went with `ExecStartPre` handling the waagent.conf stuff. I also placed our config into the patch for eventual upstreaming, so this whole `src_install` function is now gone.